fml/task_source.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_TASK_SOURCE_H_
+#define FLUTTER_FML_TASK_SOURCE_H_
+
+#include "flutter/fml/delayed_task.h"
+#include "flutter/fml/task_queue_id.h"
+
+namespace fml {
+
+class MessageLoopTaskQueues;
+
+enum TaskSourceGrade {
+  kPrimary,
+  kSecondary,
+};
+
+class TaskSource {
```

Comment:
Here and elsewhere, can we document this class and its methods? This is a significant change to the event looping infrastructure.

Reply:
Done!